data_preparation/predictor_data_prep.r

```diff
@@ -185,56 +178,132 @@ process_indicators <- function(crop, region, start_year, end_year, crop_mask_fil
     sel_shapes <- vect(file.path(AGML_ROOT, "shapefiles",
                        "shapefiles_MX",
                        "00ent_edited.shp"))
-    sel_shapes <- project(sel_shapes, "EPSG:4326")
   }
+  sel_shapes <- project(sel_shapes, "EPSG:4326")
+}
+
+process_ts_raster <- function(region, ind_filename, indicator, crop_mask_filename) {
+  sel_shapes <- get_shapes(region)
+  crop_mask = rast(crop_mask_filename)
+
+  # Handle NetCDF files.
+  # AgERA5 files (not ET0 files) are NetCDF or .nc files.
+  is_netcdf_file <- endsWith(ind_filename, ".nc")
+  if (is_netcdf_file) {
+    ind_rast <- rast(ind_filename, 1)
+  } else {
+    ind_rast <- rast(sel_files)
+  }
+
+  # Crop raster to shapes
+  ind_rast <- crop(ind_rast, sel_shapes)
```

Comment:
Check out the `win` parameter in `rast()`. You can use it to load a part of the raster, without loading the whole raster in memory, and also avoiding to do calculations for a `crop()`. That should save memory

Reply:
Added to some places. I skipped before resampling because there we might want to full raster.